spec/classes/agent_upgrade_prepare_spec.rb

```diff
@@ -0,0 +1,113 @@
+require 'spec_helper'
+
+describe 'agent_upgrade::prepare' do
+  context 'on Linux' do
+    let(:facts) {{
+      :osfamily => 'RedHat',
+      :puppet_ssldir   => '/dev/null/ssl',
+      :puppet_config   => '/dev/null/puppet.conf',
+      :mcollective_configured => false,
+    }}
+
+    [true, false].each do |mcollective_configured|
+      context "with mcollective_configured = #{mcollective_configured}" do
+        let(:facts) {{
+          :osfamily => 'RedHat',
```

Comment:
I'm overriding the `:mcollective_configured` fact; I didn't see a great way to do that while inheriting the rest of `facts`.